haystack/preview/components/samplers/top_p.py

```diff
@@ -0,0 +1,147 @@
+import logging
+from pathlib import Path
+from typing import List, Optional, Union, Dict, Any
+
+import numpy as np
+from haystack.lazy_imports import LazyImport
+from haystack.preview import Document, component, default_from_dict, default_to_dict
+
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(
+    message="Run 'pip install transformers[torch,sentencepiece]==4.32.1 sentence-transformers>=2.2.0'"
+) as torch_and_transformers_import:
+    from sentence_transformers import CrossEncoder
```

Comment:
Sorry just my curiosity: Do we strictly need to install transformers with these extras here? Can we remove it from the message maybe?